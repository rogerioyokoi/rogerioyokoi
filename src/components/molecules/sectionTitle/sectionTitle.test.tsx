import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import SectionTitle from './sectionTitle';

describe('Components > SectionTitle', () => {
  const mockSectionTitleText = 'Mock Section Title Text';
  const mockSectionTitleHighlight = 'Mock Section Title Highlight';

  describe('Snapshot', () => {
    it('Should match snapshot without Highlight', () => {
      const { container } = render(<SectionTitle text={mockSectionTitleText} />);

      expect(container).toMatchSnapshot();
    });

    it('Should match snapshot with Highlight', () => {
      const { container } = render(<SectionTitle text={mockSectionTitleText} highlight={mockSectionTitleHighlight} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests (A11y)', () => {
    it('Should not have accessibility violations without Highlight', async () => {
      const { container } = render(<SectionTitle text={mockSectionTitleText} />);

      expect(await axe(container)).toHaveNoViolations();
    });

    it('Should not have accessibility violations with Highlight', async () => {
      const { container } = render(<SectionTitle text={mockSectionTitleText} highlight={mockSectionTitleHighlight} />);

      expect(await axe(container)).toHaveNoViolations();
    });
  });

  describe('Rendering', () => {
    it('Should render text correctly', () => {
      const { getByText } = render(<SectionTitle text={mockSectionTitleText} />);

      const textContent = getByText(mockSectionTitleText);

      expect(textContent).toBeInTheDocument();
    });

    it('Should not render highlight whithout prop', () => {
      const { queryByTestId } = render(<SectionTitle text={mockSectionTitleText} />);

      const textContent = queryByTestId('section-title-highlight');

      expect(textContent).toBeNull();
    });

    it('Should render text whith highlight correctly', () => {
      const { getByText, getByTestId } = render(
        <SectionTitle text={mockSectionTitleText} highlight={mockSectionTitleHighlight} />
      );

      const textContent = getByText(mockSectionTitleText);
      const iElmHighlight = getByTestId('section-title-highlight');

      expect(textContent).toBeInTheDocument();
      expect(iElmHighlight).toBeInTheDocument();
    });
  });

  describe('CSS Classes', () => {
    it('Should apply correct classes to highlight', () => {
      const { getByTestId } = render(
        <SectionTitle text={mockSectionTitleText} highlight={mockSectionTitleHighlight} />
      );
      const iElmHighlight = getByTestId('section-title-highlight');

      expect(iElmHighlight).toHaveClass('ml-2 text-amber-500');
    });
  });
});
