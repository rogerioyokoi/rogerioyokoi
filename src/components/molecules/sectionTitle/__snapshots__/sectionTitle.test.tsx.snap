// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > SectionTitle > Snapshot > Should match snapshot with Highlight 1`] = `
<div>
  Mock Section Title Text
  <span
    class="ml-2 text-amber-500"
    data-testid="section-title-highlight"
  >
    Mock Section Title Highlight
  </span>
</div>
`;

exports[`Components > SectionTitle > Snapshot > Should match snapshot without Highlight 1`] = `
<div>
  Mock Section Title Text
</div>
`;
